usefulUtilities/botinator/assignmentClientManager.js

```diff
@@ -0,0 +1,400 @@
+        };
+    }
+
+
+    // Stop all the bots currently playing
+    function stopAllBots(){
+        availableAssignmentClientPlayers.forEach(function(ac){
+            ac.stop();
+        });
+        botCount = 0;
+        isPlaying = false;
+    }
+
+
+    // Update all the current positions for the bots
+    function updateAllBotsPosition(){
+        availableAssignmentClientPlayers.forEach(function(ac){
+            var position = getRandomLocation(contentBoundaryCorners);
+            ac.setPosition(position);
+        });
+    }
+
+    
+    // Update all the current volumes for the bots
+    function updateAllBotsVolume(){
+        availableAssignmentClientPlayers.forEach(function(ac){
+            ac.setVolume(volume);
+        });
+    }
+
+
+    // Start playing sequence to fill players with bots
+    var AC_AVAILABILITY_CHECK_MS = 1000;
+    function startSequence(){
+        // Check to see how many bots are needed
+        if (botCount >= totalNumberOfBotsNeeded) {
+            return;
+        }
+
+        if (botCount < availableAssignmentClientPlayers.length) {
+            var player = availableAssignmentClientPlayers[botCount];
+            player.play();
+            botCount++;
+
+            if (botCount >= totalNumberOfBotsNeeded) {
+                return;
+            }
+        }
+
+        Script.setTimeout(function(){
+            startSequence();
+        }, AC_AVAILABILITY_CHECK_MS);
+    }
+
+
+    // The following borrowed from Zach's push preventer script
+    // A utility function used to ensure that all of the values in "box corner 1" are less than
+    // those in "box corner 2"
+    function maybeSwapCorners(contentBoundaryCorners, dimension) {
+        var temp;
+        if (contentBoundaryCorners[0][dimension] > contentBoundaryCorners[1][dimension]) {
+            temp = contentBoundaryCorners[0][dimension];
+            contentBoundaryCorners[0][dimension] = contentBoundaryCorners[1][dimension];
+            contentBoundaryCorners[1][dimension] = temp;
+        }
+        return [contentBoundaryCorners[0][dimension], contentBoundaryCorners[1][dimension]];
+    }
+
+
+    // Ensures that all of the values in "box corner 1" are less than those in "box corner 2".
+    function fixupContentBoundaryCorners(contentBoundaryCorners) {
+        var x = maybeSwapCorners(contentBoundaryCorners, X);
+        var y = maybeSwapCorners(contentBoundaryCorners, Y);
+        var z = maybeSwapCorners(contentBoundaryCorners, Z);
+        return [
+            [x[0], y[0], z[0]],
+            [x[1], y[1], z[1]]
+        ];
+    }
+
+    // #endregion
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+    // *************************************
+    // START CONSTS_AND_VARS
+    // *************************************
+    // #region CONSTS_AND_VARS
+
+
+    var X = 0;
```

Comment:
⚠This feels dangerous to me, especially because you're doing things like `x[0]` in your code. You're relying on case sensitivity, which works for computers (most of the time), but doesn't always work for humans reading the code. Can you make a case for needing these?

Reply:
It's me being lazy.  Lol there is no other way around it.  A part of me just likes the eaze of being able to do array operations on vectors, but I get it, it's an extra step and makes the code harder to read instead of just doing the normal x, y, z vector.  I'll update.